nabla-indexer/test/indexer_tests.ts

```diff
@@ -0,0 +1,1067 @@
+
+      let backstopPool = router.backstopPool.find((backstopPool) => backstopPool.id === address(instance.backstop))!;
+      assertTrue(backstopPool !== undefined);
+      assertEq(backstopPool.reserves, "15000000000000");
+      assertEq(backstopPool.totalSupply, "15000000000000");
+
+      // next deposit
+      await instance.backstop.deposit(unit(3));
+      indexerRouters = (await readIndexerUntil(
+        (routers) =>
+          routers
+            .find((router) => router.id === address(instance.router))
+            ?.backstopPool.find((pool) => pool.id === address(instance.backstop))?.reserves === "18000000000000",
+        3000
+      ))!;
+
+      assertTrue(indexerRouters !== undefined);
+
+      router = indexerRouters.find((router) => router.id === address(instance.router))!;
+      assertTrue(router !== undefined);
+
+      backstopPool = router.backstopPool.find((backstopPool) => backstopPool.id === address(instance.backstop))!;
+      assertTrue(backstopPool !== undefined);
+      assertEq(backstopPool.reserves, "18000000000000");
+      assertEq(backstopPool.totalSupply, "18000000000000");
+
+      // withdrawal
+      await instance.backstop.withdraw(unit(6), 0);
+      indexerRouters = (await readIndexerUntil(
+        (routers) =>
+          routers
+            .find((router) => router.id === address(instance.router))
+            ?.backstopPool.find((pool) => pool.id === address(instance.backstop))?.reserves === "12000000000000",
+        3000
+      ))!;
+
+      assertTrue(indexerRouters !== undefined);
+
+      router = indexerRouters.find((router) => router.id === address(instance.router))!;
+      assertTrue(router !== undefined);
+
+      backstopPool = router.backstopPool.find((backstopPool) => backstopPool.id === address(instance.backstop))!;
+      assertTrue(backstopPool !== undefined);
+      assertEq(backstopPool.reserves, "12000000000000");
+      assertEq(backstopPool.totalSupply, "12000000000000");
+    },
+
+    async testSwaps() {
+      const instance = await createNablaInstance(unit(10), [unit(11), unit(12)]);
+      let indexerRouters = (await readIndexerUntil(
+        (routers) => routers.find((router) => router.id === address(instance.router)) !== undefined,
+        3000
+      ))!;
+      assertTrue(indexerRouters !== undefined);
+
+      await instance.router.swapExactTokensForTokens(
+        unit(2),
+        0,
+        [address(instance.assets[1]), address(instance.assets[2])],
+        tester,
+        MAX_UINT256
+      );
+
+      indexerRouters = (await readIndexerUntil(
+        (routers) =>
+          routers
+            .find((router) => router.id === address(instance.router))
+            ?.swapPools.find((pool) => pool.id === address(instance.swapPools[0]))?.reserve === "12997193504712",
+        3000
+      ))!;
+      assertTrue(indexerRouters !== undefined);
+
+      verifyIndexer(indexerRouters, instance, [
+        {
+          reserve: "12997193504712",
```

Comment:
Is there an "easy" way to calculate these values without having to use the full equation of the curve? (or at least an approximation to do a sanity check)

Reply:
It is, among others, the root of a quadratic equation. That's why this number looks so random. I think that does not count as "easy" anymore.